bitwarden_access_manager.py

```diff
@@ -0,0 +1,672 @@
+            if len(group_access_lines) > 0:
+                result = (
+                    result + "group_access = [\n" + ",\n".join(group_access_lines) + ",\n]"
+                )
+            else:
+                result = result + "group_access = []"
+        return result
+
+class BitwardenClient(NamedTuple):
+    connection: HTTPSConnection
+    bearer_token: str
+
+    @staticmethod
+    def new(client_id: str, client_secret: str) -> BitwardenClient:
+        response = requests.post(
+            "https://identity.bitwarden.com/connect/token",
+            headers={ "Content-Type": "application/x-www-form-urlencoded" },
+            data={
+                "grant_type": "client_credentials",
+                "scope": "api.organization",
+                "Accept": "application/json",
+            },
+            auth=(client_id, client_secret),
+        )
+        bearer_token = response.json()["access_token"]
+        return BitwardenClient(HTTPSConnection("api.bitwarden.com"), bearer_token)
+
+    def _http_get(self, url: str) -> HTTPResponse:
+        self.connection.request(
+            method="GET",
+            url=url,
+            headers={
+                "Accept": "application/json",
+                "Authorization": f"Bearer {self.bearer_token}",
+            },
+        )
+
+        return self.connection.getresponse()
+
+    def get_group(self, id: str) -> Any:
+        return json.load(self._http_get(f"/public/groups/{id}"))
+
+    def get_groups(self) -> Iterable[Group]:
+        data = self._http_get(f"/public/groups")
+        groups = json.load(data)
+        for group in groups["data"]:
+            yield Group(
+                id=group["id"],
+                name=group["name"],
+                access_all=group["accessAll"],
+            )
+
+    def get_collection_members(self, groups: Any) -> Iterable[MemberCollectionAccess]:
+            for group in groups:
+                group_id = group["id"]
+
+                data = self._http_get(f"/public/groups/{group_id}/member-ids")
+                member_ids = json.load(data)
+
+                for member_id in member_ids:
+                    data = self._http_get(f"/public/members/{member_id}")
+                    member = json.load(data)
+                    yield MemberCollectionAccess(
+                        name=member["name"],
+                    )
+
+    def get_collection_groups(self, groups: Any) -> Iterable[GroupCollectionAccess]:
+
+            for group in groups:
+                if group["readOnly"] == True:
+                    access = "readonly"
+                else:
+                    access = "write"
+
+                yield GroupCollectionAccess(
+                    name=self.get_group(group["id"])["name"],
+                    access=access,
+                )
+
+    def get_member_collections(self, members: Tuple[MemberCollectionAccess, ...]):
+        print("JERE")
```

Comment:
I suppose this is a debug print and this is still a work in progress? :see_no_evil:

Reply:
Yes exactly. I don't need it anymore, deleted [6329ca67](https://github.com/ChorusOne/github-access-manager/pull/10/commits/63296ca67e73d16260b0142159d1cf51a7ea2da5#diff-7bfcbcb708fa5299e4545d8f8335bc2ab16d37b5b7e70edd0262e41d040b133aL340)